### ISSUE TITLE
EDITOR not respected

### ISSUE THREAD

**Issue by cipri-tom** [rcmdnk/homebrew-file#95]
Thanks for making this much needed manager !

I've just started using it.
When I do `brew file edit` it always open up `vim`, although I have set `EDITOR=nano`. 

I believe it is related to this line: https://github.com/rcmdnk/homebrew-file/blob/dd68cd9d0db1965a93499e6ed2404de52d434c97/bin/brew-file#L897

But I'm not sure how to stop the program to debug why the environ call doesn't see the EDITOR variablel. Can you please indicate where the source file is located after installation, so that I can help debug this?

Thanks again !

**Comment by rcmdnk** [comment]
Thanks for reporting.
The problem was reproduced at my environment, too.

It works if you call `brew-file` directly:

    $ brew-file edit

then `nano` is launched if you set `export EDITOR=nano`.

But it seems Homebrew resets some environment variables when it calls subcommands,
though some variables like HOMEBREW_BREWFILE works even at `brew file`.

I will check it.

**Comment by cipri-tom** [comment]
Thank you ! Great fast response 

So `brew file` is just a convenience over `brew-file` ?

**Comment by rcmdnk** [comment]
Yes, if you have any executables named `brew-XXX` in your PATH,
then you can call it by `brew XXX` (as same as for `git` command).

But as you saw, there seems some differences which I've not noticed.

**Comment by rcmdnk** [comment]
I found that `brew` normally loads only specific variables:

> https://github.com/Homebrew/brew/blob/ef755182bbd11bf8f49b0139d089046596ab8fb7/bin/brew#L89

`EDITOR` is not loaded.

If you set:

    export HOMEBREW_NO_ENV_FILTERING=1

then all variables will be loaded and `EDITOR` work for `brew file edit`.

I don't know what will be wrong with `HOMEBREW_NO_ENV_FILTERING=1
`, but maybe anything is not suitable for specific cases.


On the other hand, `brew` loads all `HOMEBREW_XXX` variables.
Therefore, variables for brew-file ([HOMEBREW_BREWFILE_XXX](https://homebrew-file.readthedocs.io/en/latest/settings.html) are also valid.

Therefore, I added `HOMEBREW_BREWFILE_EDITOR` option.
If you set

    export HOMEBREW_BREWFILE_EDITOR=nano

then `nano` will be used even for `brew file edit`.

Another option is to use [brew-wrap](https://homebrew-file.readthedocs.io/en/latest/brew-wrap.html).
If you wrap `brew` command by `brew-wrap`,
then `brew file edit` will be changed to `brew-file edit` directly,
and `EDITOR` option is also valid.
To enable it, just add

    if [ -f $(brew --prefix)/etc/brew-wrap ];then
      source $(brew --prefix)/etc/brew-wrap
    fi

in your .bashrc or .zshrc.

Home one of them works for you!
